test/e2e/util.go

```diff
@@ -0,0 +1,269 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func waitForDeploymentStatusConditionUntilN(t *testing.T, cl client.Client, timeout time.Duration, count int, deploymentName types.NamespacedName, conditions ...appsv1.DeploymentCondition) error {
+	t.Helper()
+	ctr := 0
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		dep := &appsv1.Deployment{}
+		if err := cl.Get(context.TODO(), deploymentName, dep); err != nil {
+			t.Logf("failed to get deployment %s: %v (retrying)", deploymentName.Name, err)
+			return false, nil
+		}
+
+		expected := deploymentConditionMap(conditions...)
+		current := deploymentConditionMap(dep.Status.Conditions...)
+		if conditionsMatchExpected(expected, current) {
```

Comment:
Are you trying to ensure that the deployment conditions are stable for at least a certain period before returning true?

Reply:
Yes